### PR TITLE
Hit position

### PR DESCRIPTION
**Pull request by siilieva**
First AdvTarget tracking in place!

@olantwin, please xcheck the change in the Digi task regarding the local position coordinate used to determine the strip.
With this change I get good comparison MC points-> MC hits comparisons:
Screenshot from 2024-08-10 18-13-29

ps to all reviewers: Feel free to skip the revision of the bulk changes in the tracker - it works and passes the pre-commit tests.

### CODE REVIEW

File: shipLHC/AdvTarget.h [hunk 1]
```diff
@@ -29,6 +29,8 @@ class AdvTarget : public FairDetector
     void SetConfPar(TString name, Float_t value) { conf_floats[name] = value; }
     void SetConfPar(TString name, Int_t value) { conf_ints[name] = value; }
     void SetConfPar(TString name, TString value) { conf_strings[name] = value; }
+    /** mean position of strips associated with readout channel **/
```
Comment (olantwin) [inline shipLHC/AdvTarget.h:32]: Shouldn't it be then end-points, not the mean? Would be good to also mention whether this is opposite corners, centre line or other.

File: python/TrackingParams_AdvSND_Aug2024.xml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+           <yH_max_xz unit="cm">  40.  </yH_max_xz>
+           <yH_min_yz unit="cm"> -35.  </yH_min_yz>
+           <yH_max_yz unit="cm">  105. </yH_max_yz>
+           <N_xH_bins> 7000 </N_xH_bins>
+           <xH_min_xz unit="cm"> -100. </xH_min_xz>
+           <xH_max_xz unit="cm">  40.  </xH_max_xz>
+           <xH_min_yz unit="cm"> -35.  </xH_min_yz>
+           <xH_max_yz unit="cm">  105. </xH_max_yz>
+     </Hough_space_format>
+     <!-- Helpers to select one HT space maxima among many.
```
Comment (olantwin) [inline python/TrackingParams_AdvSND_Aug2024.xml:184]: maximum

File: python/TrackingParams_AdvSND_Aug2024.xml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+          The n-th quantile of found peaks along 'slope' axis(yH axis above) must enlose
```
Comment (olantwin) [inline python/TrackingParams_AdvSND_Aug2024.xml:185]: enclose

File: python/TrackingParams_AdvSND_Aug2024.xml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+           <yH_max_xz unit="cm">  17.  </yH_max_xz>
+           <yH_min_yz unit="cm">  0.  </yH_min_yz>
+           <yH_max_yz unit="cm">  122. </yH_max_yz>
+           <N_xH_bins> 610 </N_xH_bins>
+           <xH_min_xz unit="cm"> -105. </xH_min_xz>
+           <xH_max_xz unit="cm">  17.  </xH_max_xz>
+           <xH_min_yz unit="cm">  0.  </xH_min_yz>
+           <xH_max_yz unit="cm">  122. </xH_max_yz>
+     </Hough_space_format>
+     <!-- Helpers to select one HT space maxima among many.
```
Comment (olantwin) [inline python/TrackingParams_AdvSND_Aug2024.xml:100]: maximum

File: python/TrackingParams_AdvSND_Aug2024.xml [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+          The n-th quantile of found peaks along 'slope' axis(yH axis) must enlose
```
Comment (olantwin) [inline python/TrackingParams_AdvSND_Aug2024.xml:101]: enclose

**Review by olantwin — CHANGES_REQUESTED** [review]
Some nitpicks, looks good to me otherwise.

**Comment by olantwin** [comment]
Could you please make the analogous change for the MuFilter as well? Should be a matter of copy paste.

**Comment by siilieva** [comment]
> Could you please make the analogous change for the MuFilter as well? Should be a matter of copy paste.

what about testing? Do we know it will provide correct result. I have not tested that system.
ps. I can make a test tomorrow morning.

**Comment by olantwin** [comment]
I think a simple closure test would be good and also sufficient: are the true coordinates of the MC points contributing within between the end points returned by the geometry?

Probably won't be able to do it today, so if you could manage, that would be amazing.

**Comment by siilieva** [comment]
pushed the changes mentioned above by @olantwin  as separate commits - will reorder and squash before the final rebase onto the `main`
Fixed the AdvMuFilter strip determination and checked it using NStrips(btw 0-767) and also using the now commited 2dED:
Screenshot from 2024-08-16 19-05-29


note: the hitColour option for the display needs optimization in terms of timing performance  - can be very slow for busy events.
Screenshot from 2024-08-16 19-10-02

**Review by olantwin — APPROVED** [review]
I'd normally complain about one-line if statements and commented code in the event display, but in the interest of time, we can fix those later :)

Thanks a lot for all these fixes and improvements!

Shall I rebase as is, or do you want to squash some commits together to have a nicer history?

**Comment by siilieva** [comment]
> I'd normally complain about one-line if statements and commented code in the event display, but in the interest of time, we can fix those later :)
> 
> Thanks a lot for all these fixes and improvements!
> 
> Shall I rebase as is, or do you want to squash some commits together to have a nicer history?

I will quickly do some re-ordering and squashing :D

**Comment by olantwin** [comment]
Ok! Ping me in case another review is needed.

**Comment by siilieva** [comment]
> Ok! Ping me in case another review is needed.

It is done and I will 'rebase' now. Looks good.

**Pull request merged**